src/overlays/actors/ovl_En_Bom_Bowl_Man/z_en_bom_bowl_man.c

```diff
@@ -21,43 +50,470 @@ const ActorInit En_Bom_Bowl_Man_InitVars = {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000710, 1.0f, 0.0f, this->frameCount, 0, -10.0f);
+    this->actor.textId = 0xC0;
+    this->dialogState = 5;
+    this->actionFunc = EnBomBowMan_WaitAsleep;
+}
+
+void EnBomBowMan_WaitAsleep(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    s16 yawDiff;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->actionFunc = EnBomBowMan_TalkAsleep;
+    } else {
+        yawDiff = ABS((s16)(this->actor.yawTowardsLink - this->actor.shape.rot.y));
+
+        if (!(this->actor.xzDistFromLink > 120.0f) && (yawDiff < 0x4300)) {
+            func_8002F2CC(&this->actor, globalCtx, 120.0f);
+        }
+    }
+}
+
+void EnBomBowMan_TalkAsleep(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        globalCtx->msgCtx.msgMode = 0x37;
+        this->actionFunc = EnBomBowMan_WakeUp;
+    }
+}
+
+void EnBomBowMan_WakeUp(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    this->frameCount = (f32)SkelAnime_GetFrameCount(&D_06000080);
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000080, 1.0f, 0.0f, this->frameCount, 2, -10.0f);
+    this->eyeMode = CHU_GIRL_EYES_OPEN_SLOWLY;
+    this->actionFunc = EnBomBowMan_BlinkAwake;
+}
+
+void EnBomBowMan_BlinkAwake(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    f32 frameCount = this->skelAnime.animCurrentFrame;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (frameCount == 30.0f) {
+        this->dialogState = 5;
+        // Check for beaten Dodongo's Cavern
+        if ((gSaveContext.eventChkInf[2] & 0x20) || BREG(2)) {
+            this->actor.textId = 0xBF;
+        } else {
+            this->actor.textId = 0x7058;
+        }
+    }
+    func_8010B720(globalCtx, this->actor.textId);
+
+    if ((this->eyeTextureIndex == 0) && (this->eyeMode == CHU_GIRL_EYES_BLINK_RAPIDLY) && (this->blinkTimer == 0)) {
+        // Blink twice, then move on
+        this->eyeTextureIndex = 2;
+        this->blinkCount++;
+        if (this->blinkCount >= 3) {
+            this->actionFunc = EnBomBowMan_CheckBeatenDC;
+        }
+    }
+}
+
+void EnBomBowMan_CheckBeatenDC(EnBomBowlMan* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if ((func_8010BDBC(&globalCtx->msgCtx) == this->dialogState) && (func_80106BC8(globalCtx) != 0)) {
+        func_80106CCC(globalCtx);
+        this->frameCount = (f32)SkelAnime_GetFrameCount(&D_060072AC);
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060072AC, 1.0f, 0.0f, this->frameCount, 0, -10.0f);
+        this->eyeMode = CHU_GIRL_EYES_AWAKE;
+        this->blinkTimer = (s16)Math_Rand_ZeroFloat(60.0f) + 20;
+        // Check for beaten Dodongo's Cavern
+        if (!(gSaveContext.eventChkInf[2] & 0x20) && !BREG(2)) {
```

Comment:
I suggest having it like this for consistency with the other check
```suggestion
        if (!((gSaveContext.eventChkInf[2] & 0x20) || BREG(2))) {
```